bin/s2nc.c

```diff
@@ -447,6 +466,19 @@ int main(int argc, char *const *argv)
         case OPT_NPN:
             npn = true;
             break;
+        case OPT_BUFFERED_SEND:
+            sscanf_matched_items = sscanf(optarg, "%"SCNu32, &send_buffer_byte_size);
+            if (sscanf_matched_items != 1) {
+                fprintf(stderr, "<buffer size> must be a positive 32 bit value\n");
+                exit(1);
+            }
```

Comment:
Yes this doesn't match the other checking.

My rational was as follows: the send buffer size is unambiguously a `uint32_t`. Atoi returns an `int` which could be 32 bits. Because it is signed, it would therefore be too small to represent a `uint32_t`.

In the other cases when we use `atoi` we are down casting to `uint8_t` or `uint16_t`. Both those types are much more likely (although strictly speaking not [guaranteed](https://begriffs.com/posts/2018-11-15-c-portability.html)) to fit within an `int`.

We also use `strtoul` which has an `unsigned long` return type. This is more likely to correctly cast down to an `uint32_t` although still, not guaranteed.

If we need to be consistent I'd suggest using `strtoul`; although I'd certainly prefer to stick with `sscanf`.

Reply:
We're not as paranoid in s2nc and s2nd as we are in the library. While I'm not a fan of just doing `(uint32_t) atoi`, that's more out of concerns about negative numbers than concerns that int or long is only two bytes. But yes, strtoul (or even just atol) would be safer. I still think sscanf is overkill.

Reply:
This makes sense. I agree that being consistent should take precedence here.

Reply:
Switched to `strtoul`